### PR TITLE
Adds rewrite recipe examples plugin

### PR DESCRIPTION
**Pull request by kunli2**
This PR adds the plugin `RewriteRecipeExamplesPlugin` to register two tasks `extractRecipeExamples` and `copyExamples`.
- `extractRecipeExamples` extracts examples from a `RewriteTest` class file and generate an associated yaml file to the path `build/rewrite/examples`.
- `copyExamples` depends on task `extractRecipeExamples` and copy generated yaml files to folder `build/resource/main/META_INF/rewrite/examples` and finally into the jar.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -125,7 +133,18 @@ tasks.named<JavaCompile>("compileJava") {
...
+
 dependencies {
+    compileOnly("org.openrewrite:rewrite-java:${rewriteVersion}")
+    compileOnly("org.openrewrite:rewrite-test:${rewriteVersion}")
+    compileOnly("org.openrewrite.gradle.tooling:model:latest.release")
+
+    runtimeOnly("org.openrewrite:rewrite-test:${rewriteVersion}")
+    runtimeOnly("org.openrewrite:rewrite-java-8:${rewriteVersion}")
+    runtimeOnly("org.openrewrite:rewrite-java-11:${rewriteVersion}")
+    runtimeOnly("org.openrewrite:rewrite-java-17:${rewriteVersion}")
```
Comment (sambsnyd) [inline build.gradle.kts:146]: Why are these here? None of these should be necessary.

Comment (kunli2) [reply build.gradle.kts:145]: Do you mean lines 143-146, 4 lines of `runtimeOnly`?
because we need to perform the visitor `ExamplesExtractor` in the rewrite to extract examples during runtime?
and without it. I got some errors when running the task 
```
Caused by: java.lang.ClassNotFoundException: org.openrewrite.ExecutionContext
        ... 187 more
```
